src/controlplane/client/authenticator/authenticator.go

```diff
@@ -137,19 +161,35 @@ func (a K8sClientAuthenticator) getTLSCertificatesFromSecret(mTLSConfig *config.
 		return nil, fmt.Errorf("could not find secret %q containing TLS configuration: %w", mTLSConfig.TLSSecretName, err)
 	}
 
-	var cert, key []byte
+	keynames := certificateSecretKeys{
+		cert: opaqueCertKey,
+		key:  opaqueKeyKey,
+		ca:   opaqueCacertKey,
+	}
+
+	if secret.Type == corev1.SecretTypeTLS {
+		a.logger.Debugf("Secret %q has type %q, using standard key names", secret.Name, secret.Type)
+
+		keynames = certificateSecretKeys{
+			cert: corev1.TLSCertKey,
```

Comment:
the API denies having a SecretTypeTLS with different keys right? even in past versions I imagine. 
To avoid someone with SecretTypeTLS and our keys (for any weird reason)

Reply:
Not 100% sure if it you can submit a tls secret without the required keys, but even if you could, you shouldn't :P

In any case, the error (that should eventually get logged) explicitly says the key that we were not able to find:

https://github.com/newrelic/nri-kubernetes/blob/controlplane-tls-certs/src/controlplane/client/authenticator/authenticator.go#L187